### PR TITLE
Updates for 2023 RC Bootcamp

### PR DESCRIPTION
**Pull request by joseph-cooke**

### CODE REVIEW

File: Exercise_1_TF_env/Exercise_1.txt [hunk 1]
```diff
@@ -1,28 +1,34 @@
...
+#####Use during training###############
+srun --partition=reservation --reservation=bootcamp_gpu_2023 --gres=gpu:1 --nodes=1 --cpus-per-task=1 --pty --mem=10G --time=02:00:00 /bin/bash 
+#######################################
 
-## Loading the pre-existing tf_env environment
-module load cuda/11.1 
-module load anaconda3/2021.05
+# Build tensorflow env from scratch in your $HOME directory:
+srun --partition=gpu --nodes=1 –-cpus-per-task=1 --pty --gres=gpu:p100:1 --mem=10G --time=02:00:00 /bin/bash
+module load anaconda3/2022.05 cuda/11.2
```
Comment (manasvita) [inline Exercise_1_TF_env/Exercise_1.txt]: We need to change this to cuda/11.8 and do away with that old system-wide environment. Same for Pytorch. The revised material on RTD shows the updated version for PyTorch but needs to be updated for TF.

**Review by manasvita — CHANGES_REQUESTED** [review]
@joseph-cooke, the rest looks good but might need to test out the latest installations to update the instructions. let me know if you would need some help with that.

File: Exercise_1_TF_env/Exercise_1.txt (hunk 1 above)
Comment (joseph-cooke) [reply Exercise_1_TF_env/Exercise_1.txt]: Sorry about that. I followed the RTD for Tensorflow. I just looked over my examples and I don't use PyTorch so I just need to update the TF instructions in the scripts and slides. I will start that right away. 

I will also update the other slides not to use the shared environment and to use the one we build in the example. I did that in the first/second example but will do so for the rest. Thank you!!

Comment (joseph-cooke) [reply Exercise_1_TF_env/Exercise_1.txt]: @manasvita - I updated the instructions and have tested this. It works but it appears to be dependent on installation order to get JupyterLab to work well with it. I also write a sbatch job to install and create the environment which might be better for the workshop. Have the attendees submit the job in the beginning while we walk through the slides the environment is being built in the background for all of them. 

I am going to update the slides next but I wanted to get all of this code tested again using the new environment. I just used the installation instructions from TensorFlow plus upgrading pip (out of routine) and install JupyterLab to get it working on OOD. 

Thank you!!

File: Exercise_3_TF_sbatch/main_tf.bash [hunk 2]
```diff
@@ -16,9 +16,10 @@
 #SBATCH --cpus-per-task=1
 #SBATCH --time=00:30:00
 
-module load anaconda3/2021.05
-module load cuda/11.1
+module load anaconda3/2022.05
+module load cuda/11.8
 
-source activate /work/bootcamp/gpu_training/tf_env
+source activate TF_env
```
Comment (manasvita) [inline Exercise_3_TF_sbatch/main_tf.bash]: Either the path needs to be added here or `conda create TF_env` needs to be before `source activate TF_env`

**Review by manasvita — CHANGES_REQUESTED** [review]

**Comment by manasvita** [comment]
@joseph-cooke, we're going with either sharing the steps to build conda environment during the training or the sbatch file to do that and not really have shared conda environment available to the users. 
Also, better to point to /scratch/<username> for building conda environments during training if they don't have access to /work.

File: Exercise_1_TF_and_PyTorch_env/Exercise_1.txt [hunk 3]
```diff
@@ -41,8 +41,9 @@ conda deactivate
 
 srun --partition=gpu --gres=gpu:1 --nodes=1 --cpus-per-task=2 --mem=10GB --time=02:00:00 --pty /bin/bash ​
 module load anaconda3/2022.05 cuda/11.7
-conda create --name pytorch_env python=3.9 -y
-source activate pytorch_env
+conda create --prefix=/scratch/$USER/pytorch_env_bootcamp python=3.9 -y
+source activate /scratch/$USER/pytorch_env_bootcamp
+conda install jypyterlab -y
```
Comment (manasvita) [inline Exercise_1_TF_and_PyTorch_env/Exercise_1.txt:46]: @joseph-cooke, fyi, with pytorch, you don't need to have conda install jupyterlab prior to installing pytorch. If you look at our instructions on RTD, it is mentioned that conda install <DS-packages> can be done after installing PyTorch. I think it's more of an issue for TF. 
But, it is fine for now.

File: Exercise_1_TF_and_PyTorch_env/TF_install.bash [hunk 4]
```diff
@@ -1,14 +1,24 @@
...
 #SBATCH -N 1
 #SBATCH -c 2
 
 # Loading the modules required
 module load anaconda3/2022.05 cuda/11.8
 
-conda create --name TF_env python=3.9 -y
-source activate TF_env
+conda create --prefix=/scratch/$USER/TF_env_bootcamp python=3.9 -y
+source activate /scratch/$USER/TF_env_bootcamp
```
Comment (manasvita) [inline Exercise_1_TF_and_PyTorch_env/TF_install.bash:21]: Ran the sbatch script and it created the TF_env_bootcamp in my /scratch successfully.

File: Exercise_3_TF_sbatch/main_tf.bash [hunk 5]
```diff
@@ -19,7 +19,7 @@
 module load anaconda3/2022.05
 module load cuda/11.8
 
-source activate TF_env
+source activate /scratch/$USER/TF_env_bootcamp
```
Comment (manasvita) [inline Exercise_3_TF_sbatch/main_tf.bash:22]: This ran successfully as well and produced the desired result.

**Review by manasvita — APPROVED** [review]
All looks good.

**Pull request merged**